### PR TITLE
Removed --no-mathjax flag

### LINKED ISSUE #355: remove --no-mathjax flag (or not) ?
Mathjax should be bundled, but distrib might make it optional.

Cf discussion in https://github.com/jupyter/notebook/issues/352. 

Should we 

A) re-introduce what was removed by https://github.com/jupyter/notebook/pull/323, but with a placeholder for distributions ? 

B) Remove the --no-mathjax flag ?

---

Personally +0.5 for A) and -0.5 for B)

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #355

Let me know if anything needs to be changed and I will fix it ASAP.

giphy 3

-- 

[Edit] 
One of the question is : do we remove the `--no-mathjax` flag, that could be disable in environment that have security concern ?

### CODE REVIEW

**Comment by Carreau** [comment]
fi you superfast you can add a line into the [changelog](https://github.com/jupyter/notebook/blob/master/docs/source/changelog.rst)  for noticible features.

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,9 @@ Jupyter notebook changelog
 A summary of changes in the Jupyter notebook.
 For more detailed information, see `GitHub <https://github.com/jupyter/notebook>`__.
 
+4.1.0
+------
+- Deprecated --no-mathjax flag
```
Comment (Carreau) [inline docs/source/changelog.rst]: Bikeshedding, but isn't deprecated actually having something discouraged but still present ?

**Comment by captainsafia** [comment]
@Carreau — you are correct! Both @willingc and this cute kitty agree with you.

8325_8_site

**Comment by minrk** [comment]
I don't think we are strictly allowed to remove things in minor releases. @Carreau does `--no-mathjax` not work in 4.0?

**Comment by minrk** [comment]
PS @captainsafia issue cross-linking doesn't work in titles, so can you instead reference issues in the body in the future? That way, we get auto-closing/linking of issues on the GitHub pages.

**Comment by Carreau** [comment]
It's accepted i 4.0 but has in practice no effects and lead to the confusion/bug report of mathjax beeing made optional on Arch.

**Comment by captainsafia** [comment]
> @captainsafia issue cross-linking doesn't work in titles

Ah yes, thank you for letting me know! I usually put it in the commit messages, but I forgot this time.

**Comment by Carreau** [comment]
So @minrk, do you think we should just leave the flag for now and just issue a DeprecationWarning ?

**Comment by minrk** [comment]
The main reason I see for leaving `--no-mathjax` is that there could possibly be a deployment with security concerns about mathjax js, as there have been in the past. Re: #323, #352 I don't think it should be possible to *install* the notebook without mathjax, and such an installation should rightly be considered broken and incomplete. But it is conceivable to allow it to be *used* without mathjax. I don't feel too strongly about it, though, so if we want to remove the ability for it to run without mathjax, I'm okay with that proposal (in which case Deprecation is appropriate), though I would recommend bringing that up on the list in case there is anyone relying on it now.

**Comment by damianavila** [comment]
> But it is conceivable to allow it to be used without mathjax

This is a very valid point... I think we should left the option for these cases...

**Comment by Carreau** [comment]
> This is a very valid point... I think we should left the option for these cases...

Ok, fait enough. 

So do we want to get rid of the flag at all ? If no let's change #355 title.

**Comment by Carreau** [comment]
(bumping this PR to 5.x if we decide to remove Mathjax anyway)

**Comment by jakirkham** [comment]
> if we decide to remove Mathjax anyway

Do you just mean the flag or Mathjax entirely? If the latter, what is the alternative for future reference? Thanks.

**Comment by Carreau** [comment]
> Do you just mean the flag or Mathjax entirely?

No just the flag. We wont' remove Mathjax,

**Comment by ellisonbg** [comment]
Yeah, this should wait until 5.0.

**Comment by Carreau** [comment]
Now that we are on our way to 5.x, can we rebase than and decide wether or not we want to keep the --no-mathjax  flag ?

**Comment by Carreau** [comment]
Still needs rebase.

I'm going to go with: Go with deprecation cycle, leave the flag (that does nothing) but have it print a warning. If by 5.0 nobody has complained we can re-add it.

**Comment by captainsafia** [comment]
OK. That seems reasonable to me. I'll update tomorrow.

**Comment by Carreau** [comment]
> OK. That seems reasonable to me. I'll update tomorrow.

@captainsafia want to take care of that once you get your new laptop? 
Or should we assign an undergrad ?

**Comment by captainsafia** [comment]
Not sure when the new laptop would be coming so it might be better to give it to a new undergrad so they can learn a little bit too.

**Comment by Carreau** [comment]
Tagging as `closed-PR`, and closing as this is seem to have stalled a bit. 
This is just to keep the PR queue a bit shorter and focusing on what is actually worked on.
Feel free to reopen, (or ask for reopening if you don't have enough right to do so). 
If you were not the original author, and want to revive this, please feel free to do so. 

Also, we are doing our best to enable people to learn how to contribute to the project, and get commits in. If you are missing time or have any other issues that prevent you to work on that, please feel free to tell us, we can try to build on top of what you did. 

Thanks a lot and looking forward to see this moving forward again.

**Pull request closed without merging**